File: docs/guides/csharp/src/searchWithAnalyticsAndHeader.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SearchWithAnalyticsAndHeader
{
  async Task Main(string[] args)
  {
    try
    {
      var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));

      /*
      '94.228.178.246' should be replaced with your user's IP address.
      Depending on your stack there are multiple ways to get this information.
      */
      var ip = "94.228.178.246";
      var query = "query";

      var searchParams = new SearchParams(
        new SearchParamsObject { Query = query, Analytics = true }
      );

      await client.SearchSingleIndexAsync<Hit>(
        "<YOUR_INDEX_NAME>",
        searchParams,
        new RequestOptionBuilder().AddExtraHeader("X-Forwarded-For", ip).Build()
      );
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}

File: docs/guides/go/src/searchWithAnalyticsAndHeader.go
package main

import (
	"context"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
)

func searchWithAnalyticsAndHeader() {
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	/*
	  '94.228.178.246' should be replaced with your user's IP address.
	  Depending on your stack there are multiple ways to get this information.
	*/
	ip := "94.228.178.246"
	query := "query"

	searchParams := search.SearchParamsObjectAsSearchParams(
		search.NewSearchParamsObject().
			SetQuery(query).
			SetAnalytics(true),
	)

	_, err = client.SearchSingleIndex(context.Background(), "<YOUR_INDEX_NAME>", searchParams, search.WithHeaderParam("X-Forwarded-For", ip))
	if err != nil {
		panic(err)
	}
}
